### PR TITLE
refactor: use min deposit if higher

### PR DESCRIPTION
**Pull request by acdibble**

### CODE REVIEW

File: packages/swap/src/__tests__/integration.test.ts [hunk 1]
```diff
@@ -18,20 +18,28 @@ import { QuoteQueryParams } from '@/shared/schemas';
...
-
-    if (data.method === 'cf_ingress_egress_environment')
-      return Promise.resolve({ data: ingressEgressEnvironment('0x123') });
+    if (data.method === 'cf_environment') {
+      return Promise.resolve({
+        data: {
+          result: {
+            ...environment().result,
+            swapping: swappingEnvironment().result,
+            ingress_egress: ingressEgressEnvironment('0x123').result,
```
Comment (niklasnatter) [inline packages/swap/src/__tests__/integration.test.ts:38]: no need for these values, we can just use `environment()`

File: packages/swap/src/routes/__tests__/quote.test.ts [hunk 2]
```diff
@@ -37,15 +39,36 @@ jest.mock('@/shared/consts', () => ({
...
-    if (data.method === 'cf_swapping_environment')
-      return Promise.resolve({ data: swappingEnvironment() });
-
-    if (data.method === 'cf_ingress_egress_environment')
-      return Promise.resolve({ data: ingressEgressEnvironment('0x777') });
+    if (data.method === 'cf_environment') {
+      return Promise.resolve({
+        data: {
+          result: {
+            ...environment().result,
```
Comment (niklasnatter) [inline packages/swap/src/routes/__tests__/quote.test.ts:46]: same here but also okay to keep it i guess

File: packages/swap/src/routes/common.ts [hunk 3]
```diff
@@ -34,6 +34,7 @@ export const asyncHandler = <
     try {
       await handler(req, res, next);
     } catch (error) {
+      console.error(error);
```
Comment (niklasnatter) [inline packages/swap/src/routes/common.ts:37]: 🙌

File: packages/shared/src/rpc/utils.ts [hunk 4]
```diff
@@ -10,23 +11,38 @@ export const readAssetValue = <T>(
   return chainMinimums[asset.asset as keyof typeof chainMinimums];
 };
 
+export const readMinimumSwapAmount = (
```
Comment (niklasnatter) [inline packages/shared/src/rpc/utils.ts:14]: `getEffectiveMinimumSwapAmount`?

**Review by niklasnatter — APPROVED** [review]

File: packages/swap/src/__tests__/integration.test.ts (hunk 1 above)
Comment (acdibble) [reply packages/swap/src/__tests__/integration.test.ts:38]: yeah i was just trying to make it work fast. environment just clones the structure without injecting anything

File: packages/swap/src/routes/common.ts (hunk 3 above)
Comment (niklasnatter) [reply packages/swap/src/routes/common.ts]: i actually liked the log lol

**Pull request merged**